tests/test_csc.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import unittest
+import logging
+
+from lsst.ts import salobj
+from lsst.ts import ATHexapod
+
+STD_TIMEOUT = 15
+SHORT_TIMEOUT = 5
+
+
+class Harness:
+    def __init__(self, initial_state):
+        salobj.test_utils.set_random_lsst_dds_domain()
+        self.csc = ATHexapod.ATHexapodCSC(initial_state=initial_state)
+        self.remote = salobj.Remote(domain=self.csc.domain, name="ATHexapod", index=0)
+        self.server = ATHexapod.MockServer()
+        self.log = logging.getLogger(__name__)
+
+    async def __aenter__(self):
+        await self.server.start()
+        await self.csc.start_task
+        return self
+
+    async def __aexit__(self, *args):
+        await self.csc.close()
+        await self.server.stop()
+
+
+class CscTestCase(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_standard_state_transitions(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.STANDBY)
+
+                await harness.remote.cmd_start.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.DISABLED)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.DISABLED)
+
+                await harness.remote.cmd_enable.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.ENABLED)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.ENABLED)
+
+                await harness.remote.cmd_disable.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.DISABLED)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.DISABLED)
+
+                await harness.remote.cmd_standby.start()
+                self.assertEqual(harness.csc.summary_state, salobj.State.STANDBY)
+                state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(state.summaryState, salobj.State.STANDBY)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_apply_position_limits(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                harness.remote.evt_settingsAppliedPositionLimits.flush()

```

Comment:
This code has a race condition: if the CSC takes a bit too long to output the settingsAppliedPosition event then you won't flush the initial value.

I *strongly* suggest that you wait for the event twice (without flushing): first wait here for the initial value, which is output when the CSC starts. Then wait again below for the value from the applyPositionLimits command. You can ignore the first one if you like, though if the initial values are well known it would be nice to check them.

I have found that in general it is unwise to flush events, unless absolutely necessary. Especially in unit tests. It can be very difficult to get everything sequenced correctly, even if you put in "reasonable" delays.

(It is fine to flush telemetry because new samples are output all the time.)